File: myfunc-3.js
function mypow(base,exponent){
    var function_power={"name":"mypow","arguments":{"base":base,"exponent":exponent},"locals":{},"statements":[
`if(exponent<0) returned=1/mypow(base,-exponent)
else if(exponent==0) returned=1
else returned=mypow(base,exponent-1)*base`,
"returned"]}
    return callfunc(function_power)
}

function callfunc(function_def){
    var exitfunc=false
    for(statement of function_def.statements){
        var context=function_def.arguments
        var value = (function(context,statement){
          for(variable in context) eval(variable+"="+context[variable])
          return eval(statement)
        })(context,statement)
        if(exitfunc==true) break
    }
    return value
}
console.log(mypow(2,3))

File: myfunc-3.py
# sudo apt install python3 python3-pip python3-setuptools python3-wheel

def execf(statements,arguments):
    # ldict : locals dictionary
    ldict = {"exitfunc":False,"retval":None }
    ldict.update(arguments)
    def ret(retval=None):
        ldict["exitfunc"]=True
        ldict["retval"]=retval
    ldict["ret"]=ret
    for statement in statements:
        exec(statement,globals(),ldict)
        if ldict["exitfunc"]==True: break
    return ldict["retval"]

def def_func(function_def):
    def defined(*args):
        statements=function_def["statements"]
        arguments=dict(zip(function_def["arguments"],args))
        return execf(statements,arguments)
    globals()[function_def["name"]]=defined

function_mypow={"name":"mypow","arguments":["base","exponent"],"statements":[
"""if exponent<0: ret(1/mypow(base,-exponent))
elif exponent==0: ret(1)
else: ret(mypow(base,exponent-1)*base)"""]}

def_func(function_mypow)
print(mypow(2,3))

def_mysum={"name":"mysum","arguments":["x","y"],
"statements":["ret(x+y)"]
}

def_func(def_mysum)
print(mysum(2,3))